d3a_interface/device_validator.py

```diff
@@ -265,3 +267,85 @@ def validate_storage_device(**kwargs):
         raise D3ADeviceException(
             {"mis_configuration": [f"fit_to_limit & energy_rate_change_per_update "
                                    f"can't be set together."]})
+
+
+def _validate_rate(energy_rate):
+    if not CepSettings.ENERGY_RATE_RANGE.min <= energy_rate <= \
+           CepSettings.ENERGY_RATE_RANGE.max:
+        raise D3ADeviceException(
+            {
+                "misconfiguration": [f"energy_rate should be in between "
+                                     f"{CepSettings.ENERGY_RATE_RANGE.min} & "
+                                     f"{CepSettings.ENERGY_RATE_RANGE.max}."]})
+
+
+def _validate_rate_profile(energy_rate_profile):
+    for date, value in energy_rate_profile.items():
+        print(f"date: {date}")
+        if date == "filename":
+            continue
+        value = float(value) if type(value) == str else value
+        if not CepSettings.ENERGY_RATE_RANGE.min <= value <= CepSettings.ENERGY_RATE_RANGE.max:
+            raise D3ADeviceException(
+                {"misconfiguration": [f"energy_rate should at time: {date} be in between "
+                                      f"{CepSettings.ENERGY_RATE_RANGE.min} & "
+                                      f"{CepSettings.ENERGY_RATE_RANGE.max}."]})
+
+
+def validate_commercial_producer(**kwargs):
+    if "energy_rate" in kwargs and kwargs["energy_rate"] is not None:
+        if type(kwargs["energy_rate"]) is float or type(kwargs["energy_rate"]) is int:
```

Comment:
Better to use isinstance instead of type for checking the type, since it allows you to compare multiple types at once.